net/mux/errors_windows.go

```diff
@@ -0,0 +1,12 @@
+package mux
+
+import (
+	"errors"
+	"syscall"
+)
+
+// isPeerClosedConn returns true if the error is from the peer closing the
+// connection early.
+func isPeerClosedConnErr(err error) bool {
+	return errors.Is(err, syscall.WSAECONNRESET)
+}
```

Comment:
hmm, having to add two extra build-tagged files makes me wince a little, but the only alternative I see is:
```go
const WSAECONNRESET syscall.Errno = 10054
if errors.Is(err, io.EOF) ||
	errors.Is(err, syscall.ECONNRESET) ||
	errors.Is(err, syscall.EPIPE) ||
	errors.Is(err, syscall.EPROTOTYPE) ||
	errors.Is(err, syscall.ECONNABORTED) ||
	runtime.GOOS == "windows" && errors.Is(err, WSAECONNRESET) {
	err = ErrPeerClosedConn
}
```
which is obviously fairly gross as well.

I think using build tags is the best option here, but let's move the whole check into the function:

```go
// net/mux/errors.go

func isConnCloseErr(err error) bool {
	return errors.Is(err, io.EOF) ||
		errors.Is(err, syscall.ECONNRESET) ||
		errors.Is(err, syscall.EPIPE) ||
		errors.Is(err, syscall.EPROTOTYPE) ||
		errors.Is(err, syscall.ECONNABORTED)
}


// net/mux/errors_windows.go

func isConnCloseErr(err error) bool {
	return errors.Is(err, io.EOF) ||
		errors.Is(err, syscall.ECONNRESET) ||
		errors.Is(err, syscall.WSAECONNRESET) ||
		errors.Is(err, syscall.EPIPE) ||
		errors.Is(err, syscall.EPROTOTYPE) ||
		errors.Is(err, syscall.ECONNABORTED)
}
```